src/firefly/java/edu/caltech/ipac/table/io/VoTableReader.java

```diff
@@ -672,10 +697,10 @@ private static DataGroup convertToDataGroup(TableElement tableEl, StarTable tabl
 
         if (tableEl != null) {
             // attribute ID, ref, ucd, utype from TABLE
-            dg.setID((tableEl != null) ? getElementAttribute(tableEl, ID) : null);
-            dg.setRef((tableEl != null) ? getElementAttribute(tableEl, REF) : null);
-            dg.setUCD((tableEl != null) ? getElementAttribute(tableEl, UCD) : null);
-            dg.setUType((tableEl != null) ? getElementAttribute(tableEl, UTYPE) : null);
+            dg.addAttribute(ID,  getElementAttribute(tableEl, ID));
+            dg.addAttribute(REF, getElementAttribute(tableEl, REF));
+            dg.addAttribute(UCD, getElementAttribute(tableEl, UCD));
+            dg.addAttribute(UTYPE, getElementAttribute(tableEl, UTYPE));
```

Comment:
`dg.addAttribute(UCD` should be `dg.addAttribute(TableMeta.UCD`
because we do not want to couple our table model with votable.  FITS table may also have UCD and you don't want to have your code looks like this
`dg.addAttribute(VoTableReader.UCD, ucd)`
Of course, this comment applies to all of the above constants including the DESC below.